File: InventarioMedicamentos/Modelos/Medicine.cs
using System;

namespace InventarioMedicamentos.Modelos
{
    public class Medicine
    {
        public string Name { get; set; }
        public string Lot { get; set; }
        public DateTime ExpirationDate { get; set; }

        public Medicine(string name, string lot, DateTime expirationDate)
        {
            Name = name;
            Lot = lot;
            ExpirationDate = expirationDate;
        }

        public override string ToString()
        {
            return $"Nombre: {Name}, Lote: {Lot}, Vencimiento: {ExpirationDate:dd/MM/yyyy}";
        }
    }
}

File: Gestion_Inventario_Java/Medicine_Pharmacy_Main/Medicine.java
import java.util.Date;
import java.text.SimpleDateFormat;

public class Medicine{


    private String name;
    private String lot;
    private Date expirationDate; 

    public Medicine(String name, String lot, Date expirationDate){

        this.name = name;
        this.lot= lot; 
        this.expirationDate = expirationDate;
    
    }

    public String getName(){

        return name;
    }

    public String getLot(){

        return lot;
    }

    public Date getExpirationDate(){
        return expirationDate;

    }

    @Override 
    public String toString(){

        SimpleDateFormat sdf = new SimpleDateFormat ("dd/MM/yyyy");
        return "Nombre: " + name  + " Lote: " + lot + " Fecha de vencimiento: " + sdf.format(expirationDate);
    }
}